filebeat/harvester/registry.go

```diff
@@ -0,0 +1,69 @@
+package harvester
+
+import (
+	"sync"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+type Registry struct {
+	sync.RWMutex
+	harvesters map[uuid.UUID]Harvester
+	wg         sync.WaitGroup
+}
+
+func NewRegistry() *Registry {
+	return &Registry{
+		harvesters: map[uuid.UUID]Harvester{},
+	}
+}
+
+func (hr *Registry) add(h Harvester) {
+	hr.Lock()
+	defer hr.Unlock()
+	hr.harvesters[h.ID()] = h
+}
+
+func (hr *Registry) remove(h Harvester) {
+	hr.Lock()
+	defer hr.Unlock()
+	delete(hr.harvesters, h.ID())
+}
+
+func (hr *Registry) Stop() {
+	hr.Lock()
+	for _, hv := range hr.harvesters {
+		hr.wg.Add(1)
+		go func(h Harvester) {
+			hr.wg.Done()
+			h.Stop()
+		}(hv)
+	}
+	hr.Unlock()
+	hr.WaitForCompletion()
+}
+
+func (hr *Registry) WaitForCompletion() {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Registry.WaitForCompletion should have comment or be unexported